sdk/ai/Azure.AI.Projects/tests/Samples/Agent/Sample_Agent_Azure_Functions.cs

```diff
@@ -0,0 +1,139 @@
+            )
+        );
+        #endregion
+
+        #region Snippet:AzureFunctionsCreateAgentWithFunctionTools
+        Response<Agent> agentResponse = await client.CreateAgentAsync(
+            model: "gpt-4",
+            name: "azure-function-agent-foo",
+                instructions: "You are a helpful support agent. Use the provided function any "
+                + "time the prompt contains the string 'What would foo say?'. When you invoke "
+                + "the function, ALWAYS specify the output queue uri parameter as "
+                + $"'{storageQueueUri}/azure-function-tool-output'. Always responds with "
+                + "\"Foo says\" and then the response from the tool.",
+            tools: new List<ToolDefinition> { azureFnTool }
+            );
+        Agent agent = agentResponse.Value;
+        #endregion
+
+        Response<AgentThread> threadResponse = await client.CreateThreadAsync();
+        AgentThread thread = threadResponse.Value;
+
+        #region Snippet:AzureFunctionsHandlePollingWithRequiredAction
+        Response<ThreadMessage> messageResponse = await client.CreateMessageAsync(
+            thread.Id,
+            MessageRole.User,
+            "What is the most prevalent element in the universe? What would foo say?");
+        ThreadMessage message = messageResponse.Value;
+
+        Response<ThreadRun> runResponse = await client.CreateRunAsync(thread, agent);
+
+        do
+        {
+            await Task.Delay(TimeSpan.FromMilliseconds(500));
+            runResponse = await client.GetRunAsync(thread.Id, runResponse.Value.Id);
+        }
+        while (runResponse.Value.Status == RunStatus.Queued
+            || runResponse.Value.Status == RunStatus.InProgress
+            || runResponse.Value.Status == RunStatus.RequiresAction);
+        #endregion
+
+        Response<PageableList<ThreadMessage>> afterRunMessagesResponse
+            = await client.GetMessagesAsync(thread.Id);
+        IReadOnlyList<ThreadMessage> messages = afterRunMessagesResponse.Value.Data;
+
+        // Note: messages iterate from newest to oldest, with the messages[0] being the most recent
+        foreach (ThreadMessage threadMessage in messages)
+        {
+            Console.Write($"{threadMessage.CreatedAt:yyyy-MM-dd HH:mm:ss} - {threadMessage.Role,10}: ");
+            foreach (MessageContent contentItem in threadMessage.ContentItems)
+            {
+                if (contentItem is MessageTextContent textItem)
+                {
+                    Console.Write(textItem.Text);
+                }
+                else if (contentItem is MessageImageFileContent imageFileItem)
+                {
+                    Console.Write($"<image from ID: {imageFileItem.FileId}");
+                }
+                Console.WriteLine();
+            }
+        }
+    }
+    private class ExperimentalHeaderPolicy : HttpPipelinePolicy
```

Comment:
We already have a class that defines this header, used in the Bing Grounding tool and Azure AI Search samples. Can we reuse the [CustomHeaderPolicy](https://github.com/Azure/azure-sdk-for-net/blob/d68217644f59f19a7d162fe4887f869ca53d0fe4/sdk/ai/Azure.AI.Projects/tests/Samples/Agent/CustomHeaderPolicy.cs) class instead of defining a new one?

Reply:
Fixed.